wfcGen.py

```diff
@@ -164,85 +166,97 @@ def __init__(self, sx, sy, cells, tileSize=3):
 
     def setRandom(self):
         self.grid = []
+        self.entropySave = {}
         for y in range(self.sy):
             self.grid.append([])
             for x in range(self.sx):
                 self.grid[y].append(Cell([x, y]))
-        self.grid[random.randint(0, self.sy-1)][random.randint(0, self.sx-1)].setCell(random.randint(0, len(self.cells)-1), self.cells)
-        self.getFinalImage()
-
-    def loadEntropy(self):
-        self.entropySave = {}
-        for y in range(0, self.sy):
-            for x in range(0, self.sx):
-                self.grid[y][x].loadEntropy(self.grid)
-                if not self.grid[y][x].collapsed:
-                    entropy = self.grid[y][x].getEntropy(self.cells)
-                    self.entropySave[entropy] = self.entropySave.get(entropy, []) + [[x, y]]
+        rand = [random.randint(0, self.sy-1), random.randint(0, self.sx-1)]
+        self.grid[rand[0]][rand[1]].setCell(random.randint(0, len(self.cells)-1), self.cells)
+        self.loadPosEntropy([rand[1], rand[0]])
+        self.loadEntropyAroundPos([rand[1], rand[0]])
+
+    def loadPosEntropy(self, pos):
+        if (pos[0], pos[1]) in list(self.invertedEntropy.keys()) and pos in self.entropySave[self.invertedEntropy[(pos[0], pos[1])]]:
+            self.entropySave[self.invertedEntropy[(pos[0], pos[1])]].remove(pos)
+            if self.entropySave[self.invertedEntropy[(pos[0], pos[1])]] == []:
+                self.entropySave.pop(self.invertedEntropy[(pos[0], pos[1])], None)
+        self.grid[pos[1]][pos[0]].loadEntropy(self.grid)
+        entropy = self.grid[pos[1]][pos[0]].getEntropy(self.cells)
+        if entropy != -1:
+            self.entropySave[entropy] = self.entropySave.get(entropy, []) + [pos]
+            self.invertedEntropy[(pos[0], pos[1])] = entropy
+        else:
+            self.invertedEntropy.pop((pos[0], pos[1]), None)

```

Comment:
Here, as well, in most places you use `pos[0]` and `pos[1]` as two separate values. Does it make sense to unpack?
```python
x, y = pos
```

Reply:
Btw, here you may have the temptation to do something like this:
```python
def loadPosEntropy(self, pos):
    x, y = pos

    # ...

    # When you need them separately:
    do_something_with(x, y)

    # When you need them together
    do_together(pos)
```

However, I would highly recommend to avoid that, as the code makes it appear as if `x, y` and `pos` are two separate entities, while in fact they are the same thing in different forms. To make the code more clear, it is better to only use one of them, as follows:
```python
def loadPosEntropy(self, pos):
    x, y = pos  # Once unpacked, `pos` is never used again

    # ...

    # When you need them separately:
    do_something_with(x, y)

    # When you need them together
    do_together((x, y))
```